ZUGFeRD/EnumerationExtensions.cs

```diff
@@ -0,0 +1,48 @@
+﻿/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+using System;
+
+namespace s2industries.ZUGFeRD
+{
+    internal static class EnumerationExtensions
```

Comment:
good idea but seems not to be used in the library. Could you remove this here and put it into a second PR, e.g. demonstrating the use with one of the enum classes (ContentCodes, CountryCodes, CurrencyCodes and the likes)?

Reply:
Uhm, it's used twice, in particular the AsString() extension method as I need this for the concatenated output of PaymentTerms in XRechnung. This writes the value in upper case into the first segment.

#STORNO#TAGE=....

Reply:
See [line 955 in InvoiceDescriptor23CIIWriter.cs](https://github.com/jochenkirstaetter/ZUGFeRD-csharp/blob/50058f76b5e38972fb449a7910dad2a16423b1e9/ZUGFeRD/InvoiceDescriptor23CIIWriter.cs#L955) and [line 735 in InvoiceDescriptor20Writer.cs](https://github.com/jochenkirstaetter/ZUGFeRD-csharp/blob/50058f76b5e38972fb449a7910dad2a16423b1e9/ZUGFeRD/InvoiceDescriptor20Writer.cs#L735)

`sbPaymentNotes.Append($"#{((PaymentTermsType)paymentTerms.PaymentTermsType).AsString<PaymentTermsType>().ToUpper()}");`

Could have done similarly to the existing nested extension classes. However, it's kind of against DRY principle.

Reply:
didn't see that, sorry. Thanks for the suggestion, I might use this generic extension (might rename some functions :) ) and remove the non-DRY extensions.